### PR TITLE
fix(curriculum): regex checks for learn regex by spam filter

### LINKED ISSUE #56471: Learn Regex by Building Spam filter - Step 24 - Passing with incorrect regex
### Describe the Issue

Small error with accepted answers on this step. See screenshot. I typed in three [e3] characters in "free" and none in money, and it was accepted.

Lesson is trying to set up a spam filter that will flag the text "fr33 mon3y" or similar variations.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-regular-expressions-by-building-a-spam-filter/step-24

### Your code

```

const freeRegex = /fr[e3]|[e3]|[e3] money/i;
// this is incorrect but accepted

```

### Expected behavior

Although my answer does indeed have three instances of [e3], the correct (and only) solution for this step should be:

```

const freeRegex = /fr[e3][e3] mon[e3]y/i;
// this is correct, and also accepted

```

### Screenshots

Image

### System

User Agent is: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36

[a very old iMac 😄 ]

### Additional context

I first made a forum post about it here: [https://forum.freecodecamp.org/t/learn-regular-expressions-by-building-a-spam-filter-step-24/711716](url)

### PR DESCRIPTION
**Pull request by bulutyerli**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56471

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-regular-expressions-by-building-a-spam-filter/64233094a1293c079b5b0996.md [hunk 1]
```diff
@@ -33,6 +33,12 @@ assert.lengthOf(freeRegex.source.match(/\[(e3|3e)\]/g), 3);
 
 Your `freeRegex` should match `free money`.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-regular-expressions-by-building-a-spam-filter/64233094a1293c079b5b0996.md:34]:
```suggestion
Your `freeRegex` should match the entire phrase `free money`.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-regular-expressions-by-building-a-spam-filter/64233094a1293c079b5b0996.md [hunk 2]
```diff
@@ -33,6 +33,12 @@ assert.lengthOf(freeRegex.source.match(/\[(e3|3e)\]/g), 3);
...
 
+```js
+assert.match(freeRegex.source, /fr\[(e3|3e)\]\[(e3|3e)\] mon\[(e3|3e)\]/);
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-regular-expressions-by-building-a-spam-filter/64233094a1293c079b5b0996.md:37]:
```suggestion
assert.match(freeRegex.source, /fr\[(?:e3|3e)\]\[(?:e3|3e)\] mon\[(?:e3|3e)\]/);
```

No need for these to actually capture.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Also, the pattern test should probably come second, not first.

**Comment by ilenia-magoni** [comment]
You need to remove the 20 extra files from the PR

**Comment by ilenia-magoni** [comment]
It looks like you are having issues, feel free to ask for help in the discord server

**Comment by bulutyerli** [comment]
> It looks like you are having issues, feel free to ask for help in the discord server

yes i will check that and also i guess its better to make a new pr

**Comment by ilenia-magoni** [comment]
I suggested to ask for help to fix this PR, not to create a new one. There is always something you can do

**Pull request closed without merging**